File: app/MoviesApp/app/src/main/java/com/example/moviesapp/view/screens/moviescreen/impl/MovieDetailsViewModel.java
package com.example.moviesapp.view.screens.moviescreen.impl;

import android.annotation.SuppressLint;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.AsyncTask;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.example.moviesapp.data.movies.impl.MoviesRepo;
import com.example.moviesapp.view.screens.moviescreen.model.MovieScreenDetails;
import com.example.ui_components.webimage.model.WebImage;

import java.io.InputStream;
import java.net.URL;

import io.reactivex.android.schedulers.AndroidSchedulers;

public class MovieDetailsViewModel {

    private final MovieScreenDetails movieScreenDetails;
    private final MutableLiveData<WebImage> webImage;
    private final MutableLiveData<String> title;
    private final MoviesRepo moviesRepo;

    MovieDetailsViewModel(MovieScreenDetails movieScreenDetails) {
        this.movieScreenDetails = movieScreenDetails;
        this.webImage = new MutableLiveData<>();
        this.title = new MutableLiveData<>();
        this.moviesRepo = new MoviesRepo();
        onLoad();
    }

    private void onLoad() {
        try {
            this.webImage.setValue(new WebImage(null));
        } catch(Exception e) {
            e.printStackTrace();
        }

        moviesRepo.movie(movieScreenDetails.getId())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(movie -> {
                    title.setValue(movie.getTitle());
                    webImage.setValue(new ImageTask().execute(movie.getImageUrl()).get());
                });
    }

    public LiveData<String> title() {
        return title;
    }

    public LiveData<WebImage> image() {
        return webImage;
    }

    @SuppressLint("StaticFieldLeak")
    private class ImageTask extends AsyncTask<String, Void, WebImage> {

        @Override
        protected WebImage doInBackground(String... urls) {
            try {
                return new WebImage(bitmap(urls[0]));
            } catch(Exception e) {
                return new WebImage(null);
            }
        }
    }

    private Bitmap bitmap(String url) throws Exception {
        return BitmapFactory.decodeStream((InputStream) new URL(url).getContent());
    }
}

File: app/kotlin/MoviesApp/app/src/main/java/com/example/moviesapp/view/screens/moviescreen/impl/MovieDetailsViewModel.kt
package com.example.moviesapp.view.screens.moviescreen.impl

import android.annotation.SuppressLint
import androidx.lifecycle.MutableLiveData
import com.example.moviesapp.data.movies.impl.MoviesRepo
import com.example.moviesapp.view.screens.moviescreen.model.MovieScreenDetails
import com.example.ui_components.webimage.model.WebImage
import io.reactivex.android.schedulers.AndroidSchedulers

internal class MovieDetailsViewModel(private val movieScreenDetails: MovieScreenDetails) {

    private val webImage = MutableLiveData<WebImage>()
    private val title = MutableLiveData<String>()
    private val moviesRepo = MoviesRepo()

    init{
        onLoad()
    }

    fun title() = title

    fun image() = webImage

    @SuppressLint("CheckResult")
    private fun onLoad() {
        webImage.value = WebImage(null) // initial value
        moviesRepo.movie(movieScreenDetails.id)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe {
                    title.value = it.title
                    webImage.value = WebImage(it.imageUrl)
                }
    }
}